src/recipes/modal.ts

```diff
@@ -0,0 +1,162 @@
+import { defineSlotRecipe } from '@pandacss/dev'
+
+// default <a> color and <p> padding set in globalCss
+const modalRecipe = defineSlotRecipe({
+  slots: [
+    'root',
+    'closeX',
+    'content',
+    'title',
+    'text',
+    'actions',
+    'close',
+    // below must be styled in descendant selectors, since not in ModalComponent
+  ],
+  className: 'modal',
+  base: {
+    root: {
+      //   backgroundColor: 'white', // #333 for dark mode
+      //   boxShadow: '2px 5px 5px rgba(0, 0, 0, 0.5)',
+      //   border: 'solid 1px rgba(0, 0, 0, 0.2)', // solid 1px rgba(255, 255, 255, 0.1) for dark mode
+      //   padding: '15px',
+      color: '#e3e3e3',
+      position: 'absolute',
+      //   display: 'inline-block',
+      lineHeight: 1.5,
+      //   width: 'calc(100% - 100px)',
+      //   zIndex: 'var(--z-index-modal)',
+      // ^above are .modal styles, below are .popup
+      animation: 'fademostlyin 0.4s',
+      minWidth: '200px',
+      boxShadow: 'none',
+      border: 'none',
+      display: 'block',
+      width: '100%',
+      padding: '8%',
+      boxSizing: 'border-box',
+      zIndex: 'popup',
+      backgroundColor: 'bg',
+      transition: 'all 0.4s ease-out',
+    },
+    title: {
+      //   fontWeight: 400, from
+      fontWeight: 700,
+      marginBottom: '40px',
+      textAlign: 'center',
+    },
+    text: {
+      marginBottom: '2em',
+    },
+    actions: {
+      textAlign: 'center',
+    },
+    content: {
+      maxWidth: '40em',
+      margin: '0 auto',
+      maxHeight: 'none',
+    },
+    close: {
+      position: 'absolute',
+      top: '-5px',
+      right: '-5px',
+      fontSize: '12px',
+      verticalAlign: 'middle',
+      textAlign: 'center',
+      padding: '10px',
+      display: 'none',
+      '& span': {
+        display: 'inline-block',
+        width: '11px',
+        height: '11px',
+        color: {
+          base: 'rgba(0, 0, 0, 0.3)',
+          _dark: 'rgba(255, 255, 255, 0.3)',
+        },
+        borderColor: {
+          base: 'rgba(0, 0, 0, 0.3)',
+          _dark: 'rgba(255, 255, 255, 0.3)',
+        },
+      },
+    },
+    closeX: {
```

Comment:
We should do this as inline `css` in the `ModalComponent` since it does not have any variants. It's better to have it co-located with the component whenever possible.

I haven't gone through the other slots in this recipe, but let's inline any others that can be inlined.